python/lsst/pipe/analysis/utils.py

```diff
@@ -2412,3 +2426,30 @@ def measureRhoMetrics(rhoStat, thetaExtremum=1.0, operatorStr="<="):
         xi = rhoStat.xi  # for Rho 0
     avgRho = abs(np.average(xi[w]))
     return avgRho
+
+
+def getSchema(catalog):
+    """Helper function to determine "schema" of catalog.
+
+    This will be the list of columns if the catlaog type is a pandas DataFrame,
+    or the schema object if it is an `lsst.afw.table.SourceCatalog`.
+
+    Parameters
+    ----------
+    catalog : `lsst.afw.table.SourceCatalog` or `pandas.core.frame.DataFrame`
+        The source catalog under consideration.
+
+    Returns
+    -------
+    schema : `list` of `str` or `lsst.afw.table.Schema`
+        The schema associated with the ``catalog``.
+    """
+    if isinstance(catalog, pd.DataFrame):
+        schema = catalog.columns
+        if isinstance(schema, pd.Index):
+            schema = schema.tolist()
+    elif isinstance(catalog, pd.Series):
+        schema = [catalog.name, ]
+    else:
+        schema = catalog.schema
```

Comment:
For this commit to really be self-contained, shouldn't this return a list as well, even in the afw table case?  This would reveal the issues commented above where the schema are assumed to have methods attached to them.  Probably this will get resolved in future commits, but just wanted to note here.